src/common.js

```diff
@@ -108,3 +110,23 @@ export async function spawnIOTmp (cmd, input, args) {
     await rm(tmpDir, { recursive: true });
   }
 }
+
+export function resolveImport(input) {
+  // 'input' is an 'import specifier'. e.g. '@bytecodealliance/preview2-shim'
+  // 'import.meta.resolve' will resolve it to the actual source file URL to be imported. e.g. 'file:/${REPO}/packages/preview2-shim/lib/nodejs/index.js'
+  // 'fileURLToPath' will convert the URL to a file path. e.g. '${REPO}/packages/preview2-shim/lib/nodejs/index.js'
+  // And the last step is to navigate up to the parent directory specified by 'input'. e.g. '${REPO}/packages/preview2-shim'
+  let result = fileURLToPath(import.meta.resolve(input));
+
+  const inputBase = basename(input);
+  while (inputBase !== basename(result)) {
+    const resultDir = dirname(result);
+
+    // Short-circuit if we hit the root:
+    assert.notEqual(resultDir, result, `Unable to resolve import '${input}'`); 
+
+    result = resultDir;
+  }
+
+  return result;
```

Comment:
This is a little bit brittle since it is incompatible with custom aliasing of paths (assumes the path name matches the package name), so I'd suggest instead just using `import.meta.resolve` rules directly, rather than having this `resolveImport` abstraction:

```js
wasm_opt = path.resolve(fileURLToPath(import.meta.resolve('binaryen')), 'bin/wasm2js');
preview2_shims = path.resolve(fileURLToPath(import.meta.resolve('@bytecodealliance/preview2-shims')), '../../');
```

kind of thing.